tephra-core/src/main/java/co/cask/tephra/TransactionAdmin.java

```diff
@@ -0,0 +1,152 @@
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Joiner;
+import com.google.common.base.Splitter;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.HBaseConfiguration;
+import org.apache.twill.zookeeper.ZKClientService;
+
+import java.io.PrintStream;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Allows calling some methods on {@link TransactionManager} from command line.
+ */
+public class TransactionAdmin {
+  private static final String OPT_TRUNCATE_INVALID_TX = "--truncate-invalid-tx";
+  private static final String OPT_TRUNCATE_INVALID_TX_BEFORE = "--truncate-invalid-tx-before";
+  private static final String OPT_GET_INVALID_TX_SIZE = "--get-invalid-tx-size";
+  
+  private final PrintStream out;
+  private final PrintStream err;
+
+  public static void main(String[] args) {
+    TransactionAdmin txAdmin = new TransactionAdmin(System.out, System.err);
+    txAdmin.doMain(args, HBaseConfiguration.create());
+  }
+
+  public TransactionAdmin(PrintStream out, PrintStream err) {
+    this.out = out;
+    this.err = err;
+  }
+
+  @VisibleForTesting
+  void doMain(String[] args, Configuration conf) {
+    if (args.length < 1) {
+      printUsage();
+      return;
+    }
+
+    Injector injector = Guice.createInjector(
+      new ConfigModule(conf),
+      new ZKModule(),
+      new DiscoveryModules().getDistributedModules(),
+      new TransactionModules().getDistributedModules(),
+      new TransactionClientModule()
+    );
+
+    ZKClientService zkClient = injector.getInstance(ZKClientService.class);
+    zkClient.startAndWait();
+    
+    try {
+      TransactionSystemClient txClient = injector.getInstance(TransactionSystemClient.class);
+      String option = args[0];
+      
+      if (option.equals(OPT_TRUNCATE_INVALID_TX)) {
+        if (args.length != 2) {
+          printUsage();
+          return;
+        }
+        Set<Long> txIds = parseTxIds(args[1]);
+        if (!txIds.isEmpty()) {
+          out.println("Invalid list size before truncation: " + txClient.getInvalidSize());
+          txClient.truncateInvalidTx(txIds);
+          out.println("Invalid list size after truncation: " + txClient.getInvalidSize());
+        }
+      } else if (option.equals(OPT_TRUNCATE_INVALID_TX_BEFORE)) {
+        if (args.length != 2) {
+          printUsage();
+          return;
+        }
+        try {
+          long time = Long.parseLong(args[1]);
+          out.println("Invalid list size before truncation: " + txClient.getInvalidSize());
+          txClient.truncateInvalidTxBefore(time);
+          out.println("Invalid list size after truncation: " + txClient.getInvalidSize());
+        } catch (InvalidTruncateTimeException e) {
+          err.println(e.getMessage());
```

Comment:
Sorry, missed this before.  These error cases should return a non-zero exit code which is used for System.exit() in the main() method.  Otherwise you have no way to distinguish successful operation vs. failure when scripting.